model/atmosphere/diffusion/tests/diffusion/integration_tests/test_benchmark_diffusion.py

```diff
@@ -0,0 +1,228 @@
+        backend=backend,
+        metadata=metrics_attributes.attrs,
+        rayleigh_type=RayleighType.KLEMP,
+        rayleigh_coeff=5.0,
+        exner_expol=0.333,
+        vwind_offctr=0.2,
+    )
+
+    interpolation_state = diffusion_states.DiffusionInterpolationState(
+        e_bln_c_s=interpolation_field_source.get(interpolation_attributes.E_BLN_C_S),
+        rbf_coeff_1=interpolation_field_source.get(interpolation_attributes.RBF_VEC_COEFF_V1),
+        rbf_coeff_2=interpolation_field_source.get(interpolation_attributes.RBF_VEC_COEFF_V2),
+        geofac_div=interpolation_field_source.get(interpolation_attributes.GEOFAC_DIV),
+        geofac_n2s=interpolation_field_source.get(interpolation_attributes.GEOFAC_N2S),
+        geofac_grg_x=interpolation_field_source.get(interpolation_attributes.GEOFAC_GRG_X),
+        geofac_grg_y=interpolation_field_source.get(interpolation_attributes.GEOFAC_GRG_Y),
+        nudgecoeff_e=interpolation_field_source.get(interpolation_attributes.NUDGECOEFFS_E),
+    )
+
+    metric_state = diffusion_states.DiffusionMetricState(
+        mask_hdiff=metrics_field_source.get(metrics_attributes.MASK_HDIFF),
+        theta_ref_mc=metrics_field_source.get(metrics_attributes.THETA_REF_MC),
+        wgtfac_c=metrics_field_source.get(metrics_attributes.WGTFAC_C),
+        zd_intcoef=metrics_field_source.get(metrics_attributes.ZD_INTCOEF_DSL),
+        zd_vertoffset=metrics_field_source.get(metrics_attributes.ZD_VERTOFFSET_DSL),
+        zd_diffcoef=metrics_field_source.get(metrics_attributes.ZD_DIFFCOEF_DSL),
+    )
+    # initialization of the diagnostic and prognostic state
+    diagnostic_state = diffusion_states.DiffusionDiagnosticState(
+        hdef_ic=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+        div_ic=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+        dwdx=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+        dwdy=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+    )
+
+    prognostic_state = prognostics.PrognosticState(
+        w=data_alloc.random_field(mesh, dims.CellDim, dims.KDim, low=0.0),
+        vn=data_alloc.random_field(mesh, dims.EdgeDim, dims.KDim),
+        exner=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+        theta_v=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+        rho=data_alloc.random_field(mesh, dims.CellDim, dims.KDim),
+    )
+
+    diffusion_granule = diffusion.Diffusion(
+        grid=mesh,
+        config=config,
+        params=diffusion_parameters,
+        vertical_grid=vertical_grid,
+        metric_state=metric_state,
+        interpolation_state=interpolation_state,
+        edge_params=edge_geometry,
+        cell_params=cell_geometry,
+        backend=backend,
+        orchestration=False,
+    )
+
+    benchmark.pedantic(
+        diffusion_granule.run,
+        args=(diagnostic_state, prognostic_state, dtime),
+        rounds=10,
+        warmup_rounds=2,
+        iterations=1,
+    )
```

Comment:
Sorry, in case this was already discussed somewhere, but it's not clear to me why we use the pedantic mode. If there is a good reason, I would propose to add a comment why we need it.